### ISSUE TITLE
Unable to communicate between containers on the same node using Container Host IP address

### ISSUE THREAD

**Issue by gauravhub** [Azure/service-fabric-issues#234]
We have provisioned a Service Fabric Cluster (Win 2016 + Containers) and want to port our legacy applications on this cluster. Due to low latency requirement and the chatty communication between some of our apps, we want these apps to communicate locally. I have used Port binding to expose services hosted within containers. I then tried to use <Container Host IP Address : Port Number> to access the service and I am getting HTTP 404 errors. 

I am assuming this is due to WinNAT limitation which has been mentioned by #227

**Comment by mani-ramaswamy** [comment]
Is your question on how to specify an IP address for the cluster to work around the WinNAT issue? If so, please follow the instructions @ https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-cluster-creation-for-windows-server (1A, and you'll provide the IP address in the cluster manifest as opposed to localhost).

**Comment by gauravhub** [comment]
Hi @mani-ramaswamy, what i was referring to as a possible workaround is the ability to set IP Address or HostName for container itself. Right now containers on ASF, get a random IP Address and hostname. If we can get an ability to explicitly assign them, then we can perform inter-container communication on same host either using IP Address or hostname.

The ASF cluster which I am using in Azure already has IP address for the hosts.

**Comment by RajeetN** [comment]
Hi Gaurav,
Even though containers get an internal IP address today, you should be able to have 2 containers on the same host (or different) talk to each other using the Host IP address:HostPort. WinNAT bug should not impact this issue.

**Comment by gauravhub** [comment]
You are right @RajeetN but is there a provision to in ASF to specify IP address in ServiceManifest.xml so that containers can get a static private IP address? Also this same IP address should be assigned in a scenario when container crashes and ASF recovers it by starting a new instance.

Right now what I am observing is that containers are getting a random IP from subnet range 172.20.16.0/20.

**Comment by mani-ramaswamy** [comment]
Not yet, Gaurav - we're working on the feature to allow static IPs per container.

**Comment by gauravhub** [comment]
Thanks @mani-ramaswamy ...Will there be an option to specify hostname to be used for the container instance? Right now it is assigning container id as the hostname.

**Comment by mani-ramaswamy** [comment]
This isn't on the plan (yet), but @RajeetN  can comment more.

**Comment by jlara005** [comment]
Hello guys,

I am actually in need of deploying 2 containers to a service fabric cluster with each having a static IP address assigned in order for them to communicate between themselves (chat application). Has this been implemented yet?

**Comment by mani-ramaswamy** [comment]
Not yet - would you be able to use the DNS service instead?

**Comment by julioas09** [comment]
I would also need to be able to set the hostname of the container on deployment. This lack is quite frustrating for some applications.

**Comment by RajeetN** [comment]
@gauravhub, @julioas09 , yes you can specify a hostname for container using ContainerHostPolices. There is a hostname attribute you can use there. 
https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-securing-containers#set-up-gmsa-for-windows-containers

What value do you intend to specify there?

**Comment by mani-ramaswamy** [comment]
@julioas09 @gauravhub  Please let us know if that unblocks you.

**Comment by julioas09** [comment]
I tried to set the Hostname="example" attribute, but it returned an error when trying to deploy: "Exception while parsing XML file"
